lib/models/LifecycleConfiguration.js

```diff
@@ -0,0 +1,623 @@
+            return rules;
+        }
+        const lifecycleConf = this._parsedXML.LifecycleConfiguration;
+        const rulesArray = lifecycleConf.Rule;
+        if (!rulesArray || rulesArray.length === 0) {
+            rules.error = errors.MalformedXML.customizeDescription(
+                'request xml does not include at least one rule');
+            return rules;
+        }
+        if (rulesArray.length > 1000) {
+            rules.error = errors.MalformedXML.customizeDescription(
+                'request xml includes over max limit of 1000 rules');
+            return rules;
+        }
+        for (let i = 0; i < rulesArray.length; i++) {
+            const rule = this._parseRule(rulesArray[i]);
+            if (rule.error) {
+                rules.error = rule.error;
+            } else {
+                this._config.rules.push(rule);
+            }
+        }
+        return rules;
+    }
+
+    /**
+     * Check that each xml rule is valid
+     * @param {object} rule - a rule object from Rule array from this._parsedXml
+     * @return {object} - contains error if any component returned an error
+     * or parsing failed, else contains parsed rule object
+     *
+     * Format of ruleObj:
+     * ruleObj = {
+     *      ruleID: <value>,
+     *      ruleStatus: <value>,
+     *      filter: {
+     *          rulePrefix: <value>,
+     *          tags: [
+     *              {
+     *                  key: <value>,
+     *                  val: <value>,
+     *              }
+     *          ]
+     *      }
+     *      actions: [
+     *          {
+     *              actionName: <value>,
+     *              day: <value>,
+     *              date: <value>,
+     *              deleteMarker: <value>
+     *          },
+     *      ]
+     * }
+     */
+    _parseRule(rule) {
+        const ruleObj = {};
+        if (rule.Transition || rule.NoncurrentVersionTransition) {
+            ruleObj.error = errors.NotImplemented.customizeDescription(
+                'Transition lifecycle action not yet implemented');
+            return ruleObj;
+        }
+        if ((!rule.Filter && rule.Filter !== '') &&
+        (!rule.Prefix && rule.Prefix !== '')) {
+            ruleObj.error = errors.MalformedXML.customizeDescription(
+                'Rule xml does not include valid Filter or Prefix');
+            return ruleObj;
+        }
+        if (rule.Filter && rule.Prefix) {
+            ruleObj.error = errors.MalformedXML.customizeDescription(
+                'Rule xml should not include both Filter and Prefix');
+            return ruleObj;
+        }
+        if (!rule.Status) {
+            ruleObj.error = errors.MalformedXML.customizeDescription(
+                'Rule xml does not include Status');
+            return ruleObj;
+        }
+        const subFilter = rule.Filter ? rule.Filter[0] : rule.Prefix;
+
+        const id = this._parseID(rule.ID);
```

Comment:
just checking here because I'm not exactly sure, is `rule.ID` not an array like `Filter` or `Status`?

Reply:
It is, but it can also be totally omitted from the passed config, so I check for an array in the `_parseID` function :)